tests/unit/file_watcher_tests.py

```diff
@@ -115,3 +120,78 @@ def test_timeout(self, mock_retry_factory):
                     parser=json.load,
                     timeout=3,
                 )
+
+    @unittest.skipIf(sys.version_info.major >= 3,
```

Comment:
I think I might just break these up into separate test classes, one for Py2 and one for Py3